src/libraries/System.Text.Json/tests/System.Text.Json.SourceGeneration.Tests/JsonSerializerContextTests.cs

```diff
@@ -37,18 +37,12 @@ public static void Converters_AndTypeInfoCreator_NotRooted_WhenMetadataNotPresen
                     Assert.Contains("JsonSerializerOptions", exAsStr);
 
                     // This test uses reflection to:
-                    // - Access JsonSerializerOptions.s_defaultSimpleConverters
-                    // - Access JsonSerializerOptions.s_defaultFactoryConverters
-                    // - Access JsonSerializerOptions.s_typeInfoCreationFunc
+                    // - Access JsonSerializerOptions.s_reflectionTypeInfoResolver
                     //
                     // If any of them changes, this test will need to be kept in sync.
 
-                    // Confirm built-in converters not set.
-                    AssertFieldNull("s_defaultSimpleConverters", optionsInstance: null);
-                    AssertFieldNull("s_defaultFactoryConverters", optionsInstance: null);
-
-                    // Confirm type info dynamic creator not set.
-                    AssertFieldNull("s_typeInfoCreationFunc", optionsInstance: null);
+                    // Confirm reflection resolver not set.
+                    AssertFieldNull("s_reflectionTypeInfoResolver", optionsInstance: null);
```

Comment:
we do not need to check all fields for reflection resolver because it's not possible to create an instance and not root converters